src/ShallowTraversal.js

```diff
@@ -27,6 +27,9 @@ export function childrenOfNode(node) {
 
 export function hasClassName(node, className) {
   let classes = propsOfNode(node).className || '';
+  if (node && typeof node.type === 'function' && typeof classes !== 'string') {
```

Comment:
I don't have the overview of the whole codebase, but I would consider it good if the method throws if its a native component and `className` is not a string. That should not be a  silent error?
So;
`node && typeof node.type === 'function'` if that is true the node currently being processed is a custom component. If its not a custom component e.g a regular div, span etc, it will continue and have the same behavior as today, that is throw a error if you have passed something else then a string to `className`. 

That's my take on it. What do you think? Your decision, of course :)